views/topologicalSort.pug

```diff
@@ -1,9 +1,263 @@
-extends layout
+doctype html
+html
+  head
+    meta(charset='UTF-8')
+    title Topological Sorting
+  style.
+    .axis path,
+    .axis line{
+    fill: none;
+    stroke: black;
+    shape-rendering: crispEdges;
+    }
+    .axis text {
+    font-family: sans-serif;
+    font-size: 11px;
+    }
+  body
 
-block content
-  div(class= 'graph')
+    script(src='http://d3js.org/d3.v3.min.js', charset='utf-8')
+    script.
+        // function
+        function findPath(mid_name, vertex_dict, a, b, threLower, threUpper){
+            // mid_name: the food name of the "middle" one on the path
```

Comment:
The calculation part should be moved into the controller.

Reply:
I am afraid that the function cannot be moved into the controller. It is called when the mouse click a new point to find a new path.

Reply:
As long as the parameters are well defined, it should be ok. Since you didn't use any variable related to d3 and other frontend variables.

Reply:
btw, you don't have to move drawPath function since it is totally for views. If you want to do that, move it to public/. Also, you can't move drawPath function without any modification (since you use svg which is not defined neither in the function nor in the input parameters)

Reply:
Sure you are right about drawPath. I don't move it. But function findPath needs to be performed many times without refreshing the page. It seems js files cannot be called if they are in controllers, not public.